torch/amp/autocast_mode.py

```diff
@@ -207,6 +207,8 @@ def __init__(
             raise ValueError(
                 f"Expected `device_type` of type `str`, got: `{type(device_type)}`"
             )
+        if dtype is None:
+            dtype = torch.get_autocast_dtype(device_type)
```

Comment:
is it covered by existing UTs?

Reply:
Add two new UTs to cover this change in eager and jit path respectively.

Reply:
We should update the doc to mention the new default value for this arg?

Reply:
Updated.